include/gatekeeper_net.h

```diff
@@ -40,4 +42,14 @@ int gatekeeper_init_network(struct net_config *net_conf);
 
 void gatekeeper_free_network(void);
 
+/*
+ * Each functional block listens on one or more lcores. The lcore
+ * identifier also serves as a queue identifier.
+ */
+static inline uint16_t
+get_queue_id(void)
+{
+	return (uint16_t)rte_lcore_id();
```

Comment:
This patch is assuming that the id of the lcores are sequential and starting at zero, what's not always true. There are parameters of DPDK to force the use of the highest lcores for example.

A solution for this problem is for `get_queue_id()` to use an array whose index is the lcore id, and its values the queue id. Of course this array will need to be initialized, and `gatekeeper_init_network()` can easily do so.